src/cpp/session/SessionMain.cpp

```diff
@@ -1853,6 +1853,32 @@ void afterResume()
    LOG_DEBUG_MESSAGE("Resume complete");
 }
 
+void logStartingEnv()
+{
+#ifdef __linux__
+   LOG_DEBUG_MESSAGE("Starting R session with LD_LIBRARY_PATH: " + core::system::getenv("LD_LIBRARY_PATH"));
+#endif
+#ifdef __APPLE__
+   std::string envVars[] = {"LD_LIBRARY_PATH", "DYLD_LIBRARY_PATH", "DYLD_FALLBACK_LIBRARY_PATH"};
```

Comment:
Some thoughts:

1. Do we need to log `LD_LIBRARY_PATH`? As far as I know, that's ignored on macOS.
2. Should we just log _all_ of these environment variables? They effect the dynamic loader in different ways, and it's possible to have multiple values set.

We might also want to log `DYLD_INSERT_LIBRARIES` as well. On macOS desktop, we use this to ensure the right version of R is loaded by the R session executable; logging this could be useful to check that we're loading the version of R we expect to.